backend/src/routes/api/getUserProfileById.js

```diff
@@ -0,0 +1,22 @@
+import { StatusCodes } from 'http-status-codes';
+import Profile from '../../models/Profile.js';
+
+const userProfileEndpoint = async (router) => {
+  router.get('/users/:id', (req, res) => {
+    let id = req.params.id;
+    Profile.findById(id, function (err, profile) {
+      if (err) {
+        console.log(err);
+        res.status(StatusCodes.BAD_REQUEST).send('Something went wrong.');
+      } else {
+        if (profile === null) {
+          res.status(StatusCodes.NOT_FOUND).send('User not found.');
+        } else {
+          res.status(StatusCodes.OK).send(profile);
+        }
```

Comment:
I wonder if this can be simplified. Instead of using callback of findById we may simply 
```js
currentProfile = await Profile.findById(...);
if(profile) {
res.status(StatusCodes.OK).send(profile);
} else {
throw new Error()
} 

```

Reply:
whole code should be in try{}catch(e){} block